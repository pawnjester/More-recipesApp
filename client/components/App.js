import React from 'react';
import style from '../styles/index.scss';

export default () => {
  return (
    <h1>Hello World</h1>
  );
}