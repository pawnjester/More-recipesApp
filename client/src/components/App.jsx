import React from 'react';
import Routes from './Routes';

const App = () => (
  <div className="wrap">
    <Routes />
  </div>
);


export default App;
