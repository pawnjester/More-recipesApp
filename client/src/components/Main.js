import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Home from './homepage';
import SignUp from './signup/signUpPage';
import LoginPage from './login/loginPage';



class Main extends React.Component {
  render() {
    return (
    <main>
    <Switch>
      <Route exact path ='/' component= {Home} />
      <Route exact path ='/signup' component= {SignUp} /> 
      <Route exact path = '/signin' component ={LoginPage} />     
    </Switch>
  </main>
    )
  }
}

export default Main
