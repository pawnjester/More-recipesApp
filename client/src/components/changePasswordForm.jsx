import React, { Component } from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import NavigationBar from './NavigationBar';
import changePassword from '../actions/changePassword';
import '../styles/passwordchange.scss';

/**
 *
 *
 * @class ChangePasswordForm
 *
 * @extends {Component}
 */

class ChangePasswordForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      oldPassword: '',
      password: ''
    };
  };

  onNameChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  onSubmit = (event) => {
    event.preventDefault();
    this.props.changePassword(this.state).then((res) => {
      toastr.success('Password successfully changed');
      this.setState({
        oldPassword: '',
        password: ''
      });
    })
      .catch(error => toastr.error(error.response.data.error));
  }

  render() {
    return (
      <div >
        <NavigationBar />
        <div>
          <form className="mx-auto app-login-form card-login">
            <h4 className="black text-center mt-5">Change Your Password</h4>
            <section className="pt-4 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="oldPassword"
                className="form-control"
                type="password"
                value={this.state.oldPassword}
                onChange={this.onNameChange}
                placeholder="Enter your old password"
              />
            </section>
            <section className="pt-4 pb-3 input-group">
              <span className="input-group-addon">
                <i className="fa fa-lock black" aria-hidden="true"></i>
              </span>
              <input
                name="password"
                className="form-control"
                type="password"
                value={this.state.password}
                onChange={this.onNameChange}
                placeholder="Enter a new Password"
              />
            </section>
            <div className="pt-2">
              <button
                className="btn btn-block text-capitalize password-color"
                onClick={this.onSubmit}
              >
                change Password
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}

export default connect(null, { changePassword })(ChangePasswordForm);
