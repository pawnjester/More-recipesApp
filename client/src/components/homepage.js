import React from 'react';
import '../styles/home.scss';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';



class Home extends React.Component {
  render() {
    const {userSignupRequest} = this.props;

    return (
    <header id ="home-section">
      <div className= "dark-overlay">
      <div className= "home-inner">
        <div className ="container">
          <div className= "row">
            <div className="col-lg-8 d-none d-lg-block">
              <h1 className= "display-4" style = {{color: 'fff'}}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Create an account
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Add A Recipe 
                </div>
                
              </div>

              <div className="d-flex flex-row">
                <div className ="p-4 align-self-start" >
                  <i className="fa fa-check"></i>
                  
                </div>
                <div className= "p-4 align-self-end" style = {{color: 'fff'}}>
                  Relax, and let the world view your wonderful recipe
                </div>
                
              </div>
            </div>
            <div className="col-lg-4">
              <div className= "card bg-danger text-center card-form">
                <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                    <Signupform userSignupRequest = {userSignupRequest} />
                </div>
              </div>
            </div>
          </div>
          
        </div>
        
      </div>
      </div>
      
    </header >
    )
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
}

export default connect(null, { userSignupRequest})(Home)