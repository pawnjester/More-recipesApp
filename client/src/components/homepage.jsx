import React from 'react';
import Signupform from './signup/signupform';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { userSignupRequest } from '../actions/signupActions';
import { Button } from 'react-bootstrap';
import NavigationBar from './NavigationBar';
import '../styles/home.scss';
/*eslint-disable */


class Home extends React.Component {
  render() {
    const { userSignupRequest } = this.props;

    return (
      <div>
        <NavigationBar />
        <header id="home-section">
          <div className="dark-overlay">
            <div className="home-inner">
              <div className="container">
                <div className="row">
                  <div className="col-lg-8 d-none d-lg-block">
                <h1 className="display-4" style={{ color: 'fff' }}>Let the  world know about your <strong>Recipe</strong> in three easy steps:</h1>
                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Create an account
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end ">
                  Add A Recipe
                </div>

                </div>

                <div className="d-flex flex-row">
                  <div className="p-4 align-self-start" >
                  <i className="fa fa-check back" />

                </div>
                  <div className="p-4 align-self-end">
                  Relax, and let the world view your wonderful recipe
                </div>

                </div>
              </div>
                  <div className="col-lg-4">
                <div className="card bg-danger text-center card-form">
                  <div className="card-body">
                  <h3>Sign Up Today</h3>
                  <p>Please fill this form to register</p>
                  <Signupform userSignupRequest={userSignupRequest} />
                </div>
                </div>
              </div>
                </div>

              </div>

            </div>
          </div>

        </header >
      </div>
    );
  }
}

Home.propTypes = {
  userSignupRequest: PropTypes.func.isRequired,
};

export default connect(null, { userSignupRequest })(Home);
