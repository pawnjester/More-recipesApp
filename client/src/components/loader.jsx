import React from 'react';
import { RingLoader } from 'react-spinners';


const Loader = props => (
  <div className="sweet-loading loader-center">
    <RingLoader
      color="#e50000"
      loading={props.loading}
    />
  </div>
);

export default Loader;
