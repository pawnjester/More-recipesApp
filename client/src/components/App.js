import React from 'react';
import '../styles/index.scss';
import NavigationBar from './NavigationBar';
import Main from './Main';
import FlashMessagesList from './flash/FlashMessagesList';

class App extends React.Component {
  render() {
    return (
      <div className= 'container'>
        <NavigationBar />
        <FlashMessagesList />
        <Main />
      </div>
    );
  }
}

export default App;