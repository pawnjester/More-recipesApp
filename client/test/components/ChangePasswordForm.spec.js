import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import { ChangePasswordForm } from '../../src/components/ChangePasswordForm';


configure({ adapter: new Adapter() });

const props = {
  changePassword: jest.fn(() => Promise.resolve()),
  errors: {}
}
const setup = () => {
  const shallowWrapper = shallow(<ChangePasswordForm {...props} />);
  return {
    shallowWrapper,
  };
}
describe('Change Password Form', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onNameChange() should', () => {
  it('be called when form input changes', () => {
    const { shallowWrapper } = setup();
    const event = {
      target: { name: 'oldPassword', value: 'randomxxcom' }
    };
    const inputForm = shallowWrapper.find('.old-password-input');
    inputForm.simulate('change', event);
    expect(shallowWrapper.instance().state.oldPassword).toBe('randomxxcom');
  });
});

describe('onSubmit', () => {
  it('submit the form', () => {
    const wrapper = shallow(<ChangePasswordForm {...props} />);
    const event = {
      preventDefault: jest.fn(),
      target: {
        oldPassword: 'andela001',
        password: 'andela234'
      }
    };
    const form = wrapper.find('.password-color');
    wrapper.setState({
      oldPassword: 'andela001',
      password: 'andela234'
    });
    form.simulate('click', event);
    expect(wrapper.instance().state.oldPassword).toEqual('andela001');
  });
});
