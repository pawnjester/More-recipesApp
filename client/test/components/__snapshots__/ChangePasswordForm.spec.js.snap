// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password Form should render correctly 1`] = `
<div>
  <Connect(NavigationBar) />
  <div>
    <form
      className="mx-auto app-login-form card-login"
    >
      <h4
        className="black text-center mt-5"
      >
        Change Your Password
      </h4>
      <section
        className="pt-4 input-group"
      >
        <span
          className="input-group-addon"
        >
          <i
            aria-hidden="true"
            className="fa fa-lock black"
          />
        </span>
        <input
          className="form-control old-password-input"
          name="oldPassword"
          onChange={[Function]}
          placeholder="Enter your old password"
          type="password"
          value=""
        />
      </section>
      <section
        className="pt-4 pb-3 input-group"
      >
        <span
          className="input-group-addon"
        >
          <i
            aria-hidden="true"
            className="fa fa-lock black"
          />
        </span>
        <input
          className="form-control new-password-input"
          name="password"
          onChange={[Function]}
          placeholder="Enter a new Password"
          type="password"
          value=""
        />
      </section>
      <div
        className="pt-2"
      >
        <button
          className="btn btn-block text-capitalize password-color"
          onClick={[Function]}
        >
          change Password
        </button>
      </div>
    </form>
  </div>
</div>
`;
